charm/kafka/reactive/kafka.py

```diff
@@ -50,18 +50,18 @@ def uninstall():
 
 
 def install_snap():
-    snap_file = get_snap_file_from_charm()
-    if not snap_file:
-        snap_file = hookenv.resource_get('kafka')
-    if not snap_file:
-        hookenv.status_set('blocked', 'missing kafka snap')
-        return
     # Need to install the core snap explicit. If not, there's
     # no slots for removable-media on a bionic install.
     # Not sure if that's a snapd bug or intended behavior.
     check_call(['snap', 'install', 'core'])
-    check_call(['snap', 'install', '--dangerous', snap_file])
+
+    snap_file = get_snap_file_from_charm() or hookenv.resource_get('kafka') or 'kafka'
```

Comment:
I think this will be easier to read and debug if this oneliner is broken up into an if-elif-else. Would be good to also log how the snap was installed, so an operator can figure out how kafka actually got installed.